### PR TITLE
Feature InputValues

### PR DESCRIPTION
**Pull request by klzns**
This PR implements InputValues, previously called Attachments.

Linked workspace:
https://breno--storecomponents.myvtex.com/star-color-top/p

Depends on:
1. https://github.com/vtex-apps/store-resources/pull/54
2. https://github.com/vtex-apps/store-theme/pull/168

### CODE REVIEW

File: react/components/ProductAssemblyOptions/InputValue/BooleanInputValue.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const [state, onChange] = useInputValue(inputValueInfo)
+  const id = useInputValueId(inputValueInfo)
+
+  const handleChange = (e: any) => {
+    const value = e.target.checked
+    onChange({ value })
+  }
+
+  return (
+    <div>
```
Comment (jgfidelis) [inline react/components/ProductAssemblyOptions/InputValue/BooleanInputValue.tsx]: empty div, is this desired?

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemInputValues.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import OptionsInputValue from './InputValue/OptionsInputValue'
+import BooleanInputValue from './InputValue/BooleanInputValue'
+
+const ProductAssemblyOptionItemInputValues: FC = () => {
+  const { inputValues } = useProductAssemblyGroupState() as AssemblyOptionGroupType
+
+  return (
+    <Fragment>
+      {inputValues.map((inputValue, index) => {
+        if (inputValue.type === 'TEXT') {
```
Comment (jgfidelis) [inline react/components/ProductAssemblyOptions/ProductAssemblyOptionItemInputValues.tsx]: create an enum of this `type`, like `InputValueType`

File: react/typings/ItemMetadata.d.ts [hunk 3]
```diff
@@ -21,17 +21,31 @@ interface AssemblyOption {
...
   composition: Composition | null
-  inputValues: AttachmentField[]
+  inputValues: InputValue[]
 }
 
-interface AttachmentField {
+type InputValue = TextInputValue | BooleanInputValue | OptionsInputValue
+
+interface TextInputValue {
+  type: 'TEXT'|string
```
Comment (victorhmp) [inline react/typings/ItemMetadata.d.ts]: Why do you need to define `type` as being `'TEXT' | string`, and not just `'TEXT'`? Other string values would be ok?

File: react/components/ProductAssemblyContext/Group.tsx [hunk 4]
```diff
@@ -72,6 +88,17 @@ export const useProductAssemblyGroupState = () =>
...
+    case 'SET_INPUT_VALUE': {
+      const { groupPath, inputValue, inputValueLabel } = action.args
+      let groupState = path(groupPath, state) as AssemblyOptionGroupType
+
+      groupState.valuesOfInputValues = {
+        ...groupState.valuesOfInputValues,
+        [inputValueLabel]: inputValue,
+      }
+
+      return { ...state }
```
[the remark this answers is not in the document]
Comment (jgfidelis) [reply react/components/ProductAssemblyContext/Group.tsx]: `groupState` is a ref to a property in the state.

Like: `state = {a:{b:{c:1}}}; let b = state.a.b; b = {c:2}; console.log(state); //state={a:{b:{c:2}}}`

Can you see how we are mutating the state, after all?

File: react/components/ProductAssemblyOptions/InputValue/BooleanInputValue.tsx (hunk 1 above)
Comment (klzns) [reply react/components/ProductAssemblyOptions/InputValue/BooleanInputValue.tsx]: I will come back to the layout later

[the remark this answers is not in the document]
Comment (klzns) [reply react/components/ProductAssemblyOptions/InputValue/BooleanInputValue.tsx]: I will come back to the layout later and change those divs

File: react/typings/ItemMetadata.d.ts (hunk 3 above)
Comment (klzns) [reply react/typings/ItemMetadata.d.ts]: Changed

File: react/components/ProductAssemblyContext/Group.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/components/ProductAssemblyContext/Group.tsx]: Oooh! I see! Makes sense hahahah

Thanks! =D

**Pull request merged**